### PR TITLE
improvements to structure log events from PID1

### PR DESCRIPTION
**Pull request by poettering**
This cleans up generation of existing structured log events a bit in PID 1 and adds three new ones.

Fixes: #10265

### CODE REVIEW

File: src/core/job.c [hunk 1]
```diff
@@ -509,6 +509,93 @@ static void job_change_type(Job *j, JobType newtype) {
...
+_pure_ static const char* job_get_begin_status_message_format(Unit *u, JobType t) {
+        const char *format;
+        const UnitStatusMessageFormats *format_table;
+
+        assert(u);
+        assert(IN_SET(t, JOB_START, JOB_STOP, JOB_RELOAD));
+
+        if (t != JOB_RELOAD) {
+                format_table = &UNIT_VTABLE(u)->status_message_formats;
+                if (format_table) {
```
Comment (keszybz) [inline src/core/job.c]: This is a pointer to a field, so it cannot be zero. This whole variable can go away, and `UNIT_VTABLE(u)->status_message_formats->starting_stopping[t == JOB_STOP]` can be accessed directly.

File: src/systemd/sd-messages.h [hunk 2]
```diff
@@ -81,16 +81,19 @@ _SD_BEGIN_DECLARATIONS;
...
 #define SD_MESSAGE_UNIT_STARTED           SD_ID128_MAKE(39,f5,34,79,d3,a0,45,ac,8e,11,78,62,48,23,1f,bf)
 #define SD_MESSAGE_UNIT_STARTED_STR       SD_ID128_MAKE_STR(39,f5,34,79,d3,a0,45,ac,8e,11,78,62,48,23,1f,bf)
+#define SD_MESSAGE_UNIT_FAILED            SD_ID128_MAKE(be,02,cf,68,55,d2,42,8b,a4,0d,f7,e9,d0,22,f0,3d)
+#define SD_MESSAGE_UNIT_FAILED_STR        SD_ID128_MAKE_STR(be,02,cf,68,55,d2,42,8b,a4,0d,f7,e9,d0,22,f0,3d)
 #define SD_MESSAGE_UNIT_STOPPING          SD_ID128_MAKE(de,5b,42,6a,63,be,47,a7,b6,ac,3e,aa,c8,2e,2f,6f)
 #define SD_MESSAGE_UNIT_STOPPING_STR      SD_ID128_MAKE_STR(de,5b,42,6a,63,be,47,a7,b6,ac,3e,aa,c8,2e,2f,6f)
 #define SD_MESSAGE_UNIT_STOPPED           SD_ID128_MAKE(9d,1a,aa,27,d6,01,40,bd,96,36,54,38,aa,d2,02,86)
 #define SD_MESSAGE_UNIT_STOPPED_STR       SD_ID128_MAKE_STR(9d,1a,aa,27,d6,01,40,bd,96,36,54,38,aa,d2,02,86)
-#define SD_MESSAGE_UNIT_FAILED            SD_ID128_MAKE(be,02,cf,68,55,d2,42,8b,a4,0d,f7,e9,d0,22,f0,3d)
-#define SD_MESSAGE_UNIT_FAILED_STR        SD_ID128_MAKE_STR(be,02,cf,68,55,d2,42,8b,a4,0d,f7,e9,d0,22,f0,3d)
```
Comment (keszybz) [inline src/systemd/sd-messages.h:93]: What about renaming the macros? We add and remvoe SD_MESSAGE_ defines without keeping backwards compat.

File: src/core/job.c [hunk 3]
```diff
@@ -877,6 +881,20 @@ static void job_log_done_status_message(Unit *u, uint32_t job_id, JobType t, Job
         if (log_on_console() && job_print_done_status_messages[result].word)
                 return;
 
+        /* No message if the job did not actually do anything due to failed condition. */
+        if (t == JOB_START && result == JOB_DONE && !u->condition_result) {
+                log_struct(LOG_INFO,
+                           "MESSAGE=Condition check resulted in %s to be skipped.", unit_description(u),
```
Comment (keszybz) [inline src/core/job.c]: "Condition check resulted in %s being skipped"

File: src/systemd/sd-messages.h (hunk 2 above)
Comment (poettering) [reply src/systemd/sd-messages.h:93]: they are installed publicly, they are part of the API I fear.

Comment (keszybz) [reply src/systemd/sd-messages.h:93]: You're right, we should not remove them.

Nevertheless, we *did* remove items from here in the past, c.f. e.g. 232c84b2d22f2d96982b3c0390d29498bb430835. It was a mistake to do this.

File: src/core/job.c (hunk 1 above)
Comment (poettering) [reply src/core/job.c]: oh, true, i guess i we can simplify this even further

File: src/systemd/sd-messages.h (hunk 2 above)
Comment (poettering) [reply src/systemd/sd-messages.h:93]: yupp, we fucked up there.

**Comment by poettering** [comment]
Force pushed a new version. Only changes are the suggested string change and a new patch that simplifies the format string extraction as requested. I did that in a separate patch to make it easier to review, and because this actually needed fixing at two places, and is about pre-existing code.

Anyway, PTAL.

**Pull request merged**